modules/azure/compute.go

```diff
@@ -8,97 +8,451 @@ import (
 // This function would fail the test if there is an error.
-func GetSizeOfVirtualMachine(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) compute.VirtualMachineSizeTypes {
-	size, err := GetSizeOfVirtualMachineE(t, vmName, resGroupName, subscriptionID)
+func GetVirtualMachineManagedDiskCount(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) int {
+	mngDiskCount, err := GetVirtualMachineManagedDiskCountE(vmName, resGroupName, subscriptionID)
 	require.NoError(t, err)
 
-	return size
+	return mngDiskCount
 }
 
-// GetSizeOfVirtualMachineE gets the size type of the given Azure Virtual Machine
-func GetSizeOfVirtualMachineE(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) (compute.VirtualMachineSizeTypes, error) {
-	// Validate resource group name and subscription ID
-	resGroupName, err := getTargetAzureResourceGroupName(resGroupName)
+// GetVirtualMachineManagedDiskCountE gets the Managed Disk count of the specified Azure Virtual Machine.
+func GetVirtualMachineManagedDiskCountE(vmName string, resGroupName string, subscriptionID string) (int, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return -1, err
+	}
+
+	return len(*vm.StorageProfile.DataDisks), nil
+}
+
+// GetVirtualMachineOsDiskName gets the OS Disk name of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineOsDiskName(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) string {
+	osDiskName, err := GetVirtualMachineOsDiskNameE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return osDiskName
+}
+
+// GetVirtualMachineOsDiskNameE gets the OS Disk name of the specified Azure Virtual Machine.
+func GetVirtualMachineOsDiskNameE(vmName string, resGroupName string, subscriptionID string) (string, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return "", err
 	}
 
-	// Create a VM client
-	vmClient, err := GetVirtualMachineClientE(subscriptionID)
+	return *vm.StorageProfile.OsDisk.Name, nil
+}
+
+// GetVirtualMachineState gets the State of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineState(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) string {
+	vmState, err := GetVirtualMachineStateE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return vmState
+}
+
+// GetVirtualMachineStateE gets the State of the specified Azure Virtual Machine.
+func GetVirtualMachineStateE(vmName string, resGroupName string, subscriptionID string) (string, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	return vm.Status, nil
+}
```

Comment:
NIT: I think this function is a bit extraneous, given that one can get this with `GetVirtualMachine(vmName, resGroupName, subscriptionID).Status`

As a rule of thumb, a single attribute probably doesn't need a getter, while nested attributes would be useful (e.g., like the `GetVirtualMachineOsDiskName` function)

Reply:
Removed! Good point.